server/src/main/java/org/apache/druid/segment/loading/SegmentLocalCacheManager.java

```diff
@@ -190,22 +193,47 @@ public List<DataSegment> getCachedSegments() throws IOException
       );
     }
 
-    final List<DataSegment> cachedSegments = new ArrayList<>();
     final File[] segmentsToLoad = retrieveSegmentMetadataFiles();
-
+    final ConcurrentLinkedQueue<DataSegment> cachedSegments = new ConcurrentLinkedQueue<>();
     AtomicInteger ignoredFilesCounter = new AtomicInteger(0);
+    CountDownLatch latch = new CountDownLatch(segmentsToLoad.length);
 
-    for (int i = 0; i < segmentsToLoad.length; i++) {
-      final File file = segmentsToLoad[i];
-      log.info("Loading segment cache file [%d/%d][%s].", i + 1, segmentsToLoad.length, file);
```

Comment:
Curious, why was this info log removed? When an operator searches for a segment ID, it'd show the entire lifecycle of the segment - from loading on a data server to being dropped, so IMO this log line is useful to keep since it’s emitted only once and helps track progress as well. If it’s not of interest, we could always configure log4j to filter out these log lines instead.